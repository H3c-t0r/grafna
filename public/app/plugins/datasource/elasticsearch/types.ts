export enum MetricFindQueryTypes {
  Fields = 'fields',
  Terms = 'terms',
}

export enum FieldTypes {
  Any = '',
  Number = 'number',
  Date = 'date',
  String = 'string',
  Keyword = 'keyword',
  Nested = 'nested',
}
