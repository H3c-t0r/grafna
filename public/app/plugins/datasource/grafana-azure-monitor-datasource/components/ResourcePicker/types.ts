export enum ResourceRowType {
  Subscription = 'Subscription',
  ResourceGroup = 'ResourceGroup',
  Resource = 'Resource',
  VariableGroup = 'TemplateVariableGroup',
  Variable = 'TemplateVariable',
}

export interface ResourceRow {
  id: string;
  name: string;
  type: ResourceRowType;
  typeLabel: string;
  location?: string;
  children?: ResourceRowGroup;
  loading?: boolean;
  expanded?: boolean;
  selected?: boolean;
}

export type ResourceRowGroup = ResourceRow[];

export interface ResourceRowData {
  resourceRowGroup: ResourceRowGroup;
}
