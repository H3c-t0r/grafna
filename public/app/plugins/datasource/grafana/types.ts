import { AnnotationQuery, DataQuery } from '@grafana/data';

//----------------------------------------------
// Query
//----------------------------------------------

export enum GrafanaQueryType {
  RandomWalk = 'randomWalk',
  RandomStream = 'randomStream',
  HostMetrics = 'hostmetrics',
}

export interface GrafanaQuery extends DataQuery {
  queryType: GrafanaQueryType; // RandomWalk by default
}

export const defaultQuery: GrafanaQuery = {
  refId: 'A',
  queryType: GrafanaQueryType.RandomWalk,
};

//----------------------------------------------
// Annotations
//----------------------------------------------

export enum GrafanaAnnotationType {
  Dashboard = 'dashboard',
  Tags = 'tags',
}

export interface GrafanaAnnotaitonQuery extends AnnotationQuery<GrafanaQuery> {
  type: GrafanaAnnotationType; // tags
  limit: number; // 100
}
