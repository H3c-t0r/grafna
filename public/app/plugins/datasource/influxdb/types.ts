import { DataQuery, DataSourceJsonData } from '@grafana/data';

export enum InfluxVersion {
  V1x = '1x',
  V2x = '2x',
}

export interface InfluxOptions extends DataSourceJsonData {
  version?: InfluxVersion;

  timeInterval: string;
  httpMode: string;

  // Influx 2.0
  organization?: string;
  defaultBucket?: string;
  maxSeries?: number;
}

export interface InfluxSecureJsonData {
  // Influx 2.0
  token?: string;

  // In 1x a different password can be sent than then HTTP auth
  password?: string;
}

export interface InfluxQueryPart {
  type: string;
  params?: string[];
  interval?: string;
}

export interface InfluxQueryTag {
  key: string;
  operator?: string;
  condition?: string;
  value: string;
}

export enum InfluxQueryType {
  Classic = 'Classic', // InfluxQL query builder
  InfluxQL = 'InfluxQL', // raw InfluxQL
  Flux = 'Flux',
}

export interface InfluxQuery extends DataQuery {
  queryType?: InfluxQueryType;
  policy?: string;
  measurement?: string;
  resultFormat?: 'time_series' | 'table';
  orderByTime?: string;
  tags?: InfluxQueryTag[];
  groupBy?: InfluxQueryPart[];
  select?: InfluxQueryPart[][];
  limit?: string;
  slimit?: string;
  tz?: string;
  fill?: string;
  rawQuery?: boolean; // deprecated (use raw InfluxQL)
  query?: string;
}
