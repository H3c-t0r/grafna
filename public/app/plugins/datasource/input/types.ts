import { DataQuery, SeriesData } from '@grafana/ui/src/types';

export interface InputQuery extends DataQuery {
  // Save data in the panel
  data?: SeriesData[];
}
