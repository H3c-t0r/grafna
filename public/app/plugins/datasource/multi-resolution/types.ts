import { DataQuery } from '@grafana/ui';

export enum ResolutionSelection {
  range = 'range',
  interval = 'interval',
}

export interface QueriesForResolution {
  resolution?: string;
  ms: number; // Will be -Infinity when resolution is not set
  targets: DataQuery[];
  datasource?: string; // Datasource to apply by default
}

export interface MultiResolutionQuery extends DataQuery {
  live?: DataQuery; // Will get passed existing panel data
  select: ResolutionSelection;
  resolutions: QueriesForResolution[];
}
