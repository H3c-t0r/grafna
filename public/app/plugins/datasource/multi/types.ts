import { DataQuery } from '@grafana/ui';

export enum ResolutionSelection {
  range = 'range',
  interval = 'interval',
}

export interface QueriesForResolution {
  id: string; // Unique ID
  ms: number; // First value will be 0
  now?: boolean; // only if to=now
  targets: DataQuery[]; // Always mixed
}

export interface MultiResolutionQuery extends DataQuery {
  live?: DataQuery; // Will get passed existing panel data
  select: ResolutionSelection;
  resolutions: QueriesForResolution[];
}
