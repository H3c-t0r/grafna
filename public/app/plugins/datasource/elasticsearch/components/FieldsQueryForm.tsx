import React, { PureComponent } from 'react';
import { Select, SelectOptionItem, FormLabel } from '@grafana/ui';
import { FieldTypes } from '../types';

export interface FieldsQueryFormProps {
  query: any;
  onChange: (query: any, definition: string) => void;
}

export interface FieldsQueryFormState {
  type: string;
}

const defaultState: FieldsQueryFormState = {
  type: '',
};

const fieldTypes: SelectOptionItem[] = [
  { value: FieldTypes.Any, label: 'Any' },
  { value: FieldTypes.Number, label: 'Number' },
  { value: FieldTypes.Date, label: 'Date' },
  { value: FieldTypes.String, label: 'String' },
  { value: FieldTypes.Keyword, label: 'Keyword' },
  { value: FieldTypes.Nested, label: 'Nested' },
];

export class FieldsQueryForm extends PureComponent<FieldsQueryFormProps, FieldsQueryFormState> {
  constructor(props: FieldsQueryFormProps) {
    super(props);
    this.state = {
      ...defaultState,
      ...props.query,
    };
  }

  componentDidMount() {
    if (!this.props.query.type) {
      this.triggerChange();
    }
  }

  triggerChange() {
    const { onChange } = this.props;
    const { type } = this.state;
    const query: any = {
      find: 'fields',
    };
    const selectedtype = fieldTypes.find(o => o.value === type);

    if (defaultState.type !== type) {
      query.type = type;
    }

    onChange(query, `Fields(${selectedtype.label})`);
  }

  onFieldTypeChange = (item: SelectOptionItem) => {
    this.setState(
      {
        type: item.value,
      },
      () => this.triggerChange()
    );
  };

  render() {
    const { type } = this.state;
    return (
      <>
        <div className="form-field">
          <FormLabel className="query-keyword">Field Type</FormLabel>
          <Select
            placeholder="Choose field type"
            isSearchable={false}
            options={fieldTypes}
            value={fieldTypes.find(o => o.value === type)}
            onChange={this.onFieldTypeChange}
            width={11}
          />
        </div>
      </>
    );
  }
}
