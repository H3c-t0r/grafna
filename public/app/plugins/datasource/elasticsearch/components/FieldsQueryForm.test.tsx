import React from 'react';
import { shallow } from 'enzyme';
import { FieldsQueryForm } from './FieldsQueryForm';
import { FieldTypes } from '../types';

const setup = (propOverrides?: object) => {
  const props = {
    onChange: () => {},
    query: {},
    ...propOverrides,
  };

  const wrapper = shallow(<FieldsQueryForm {...props} />);
  const instance = wrapper.instance() as FieldsQueryForm;

  return {
    wrapper,
    instance,
  };
};

describe('FieldsQueryForm', () => {
  it('should render component', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('when loading component with type should not trigger change event', () => {
    const props = {
      onChange: jest.fn(),
      query: { type: 'keyword' },
    };
    setup(props);
    expect(props.onChange.mock.calls.length).toBe(0);
  });

  it('when loading component without type should trigger change event', () => {
    const props = {
      onChange: jest.fn(),
    };
    setup(props);
    expect(props.onChange.mock.calls.length).toBe(1);
    expect(props.onChange.mock.calls[0][0].find).toBe('fields');
    expect(props.onChange.mock.calls[0][1]).toBe('Fields(Any)');
  });

  it('when changing type should trigger change event', () => {
    const props = {
      onChange: jest.fn(),
    };
    const { instance } = setup(props);
    props.onChange.mockReset();
    instance.onFieldTypeChange(FieldTypes.Keyword);
    expect(props.onChange.mock.calls.length).toBe(1);
    expect(props.onChange.mock.calls[0][0].find).toBe('fields');
    expect(props.onChange.mock.calls[0][0].type).toBe(FieldTypes.Keyword);
    expect(props.onChange.mock.calls[0][1]).toBe('Fields(Keyword)');
  });

  it('when changing type to template variable should trigger change event', () => {
    const props = {
      onChange: jest.fn(),
    };
    const { instance } = setup(props);
    props.onChange.mockReset();
    instance.onFieldTypeChange('$var');
    expect(props.onChange.mock.calls.length).toBe(1);
    expect(props.onChange.mock.calls[0][0].find).toBe('fields');
    expect(props.onChange.mock.calls[0][0].type).toBe('$var');
    expect(props.onChange.mock.calls[0][1]).toBe('Fields($var)');
  });
});
