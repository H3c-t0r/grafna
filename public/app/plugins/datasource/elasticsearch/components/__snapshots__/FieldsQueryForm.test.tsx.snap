// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldsQueryForm should render component 1`] = `
<Fragment>
  <div
    className="form-field"
  >
    <Component
      className="query-keyword"
    >
      Field Type
    </Component>
    <Select
      autoFocus={false}
      backspaceRemovesValue={true}
      className=""
      components={
        Object {
          "Group": [Function],
          "IndicatorsContainer": [Function],
          "MenuList": [Function],
          "Option": [Function],
          "SingleValue": [Function],
        }
      }
      isClearable={false}
      isDisabled={false}
      isLoading={false}
      isMulti={false}
      isSearchable={false}
      maxMenuHeight={300}
      menuIsOpen={false}
      onChange={[Function]}
      openMenuOnFocus={false}
      options={
        Array [
          Object {
            "label": "Any",
            "value": "",
          },
          Object {
            "label": "Number",
            "value": "number",
          },
          Object {
            "label": "Date",
            "value": "date",
          },
          Object {
            "label": "String",
            "value": "string",
          },
          Object {
            "label": "Keyword",
            "value": "keyword",
          },
          Object {
            "label": "Nested",
            "value": "nested",
          },
        ]
      }
      placeholder="Choose field type"
      value={
        Object {
          "label": "Any",
          "value": "",
        }
      }
      width={11}
    />
  </div>
</Fragment>
`;
